### ISSUE TITLE
Files corrupted on clone (urls inside a file, not data)

### ISSUE THREAD

**Issue by alconost** [github/git-lfs#678]
**I'm cloning a repository and get the following error (from the logs):**

$ git-lfs.exe smudge fonts/GOTHICBI.TTF
Error accessing media: fonts/GOTHICBI.TTF (2dc5949d57d2e172601fb6f5093c1fbf15a46
3e29ed47c4c8ff2434baf1c2b19)

Error for GET https://api.github.com/lfs/alconost/maki_lfs/objects/2dc5949d57d2e
172601fb6f5093c1fbf15a463e29ed47c4c8ff2434baf1c2b19
goroutine 1 [running]:
github.com/github/git-lfs/lfs.Stack(0x0, 0x0, 0x0)
        C:/Users/techn/go/src/github.com/github/git-lfs/lfs/errors.go:525 +0x87
...

**The file GOTHICBI.TTF that appears on the disk contais an url, not data:**

version https://git-lfs.github.com/spec/v1
oid sha256:2dc5949d57d2e172601fb6f5093c1fbf15a463e29ed47c4c8ff2434baf1c2b19
size 139084

**So, the file is corrupted and clone\pull does not resume on errors...**

**Comment by technoweenie** [comment]
So what's happening is that there was an error downloading the file through the LFS API, so the "LFS pointer" object that Git LFS leaves in the Git repository is still in your working directory. I did confirm that the object is still on our servers, so it doesn't look like there's any corrupt data.

To help you further, I'll need full details of the error. You should be able to get this by running `git lfs logs last` in your repository. It looks like this is a private repository, so feel free to send more details to https://github.com/contact so I can help you privately.

Now, once the download problem is figured out, you can fix this easily locally:

1. Ensure you're on the latest Git LFS client: `git lfs version` should say v0.6.0.
2. Run `git lfs pull` to try to download the files again.

**Comment by alconost** [comment]
Hi!

Thanks for the reply. Yes, indeed, the file itself is ok on your server (if retried, it downloads succesfully). git lfs pull helped.

But, there is a serious problem that prevents us from using git lfs now. If during pull we get a file with lfs pointer instead of e.g. WAV file, there is a risk that we release our software packaged with that "corrupt file" instead of .WAV. It won't work.

If there is error downloading files during initial **clone**, then **git pull** will not update files.

We have a 10GB repo with 80k files. Some of them are 300MB. I never managed to clone the repo because of the download errors. It seems it does not retry on download failures.

Is there any progress and understanding that there should be retries on donwload errors? Amazon S3  storage does not seem to be very reliable regarding connectivity, unfortunately.

Thank you!

**Comment by technoweenie** [comment]
The Git smudge filter only works on checkout. So if there's a download error, `git pull` won't update bare LFS pointer files unless those files change in the pulled commits. `git lfs pull` is what you'll want to ensure all the LFS pointers are updated.

v0.6.0 adds retries, but not resumable retries. We're tracking more things to improve in https://github.com/github/git-lfs/issues/414.

**Comment by pbrooks** [comment]
Hi guys,
I've recently come across this issue with my local copy, other users seem to have this file in place.

I believe it's occured when I've cancelled the lfs pull by using a signal. It's left the metadata file instead of the actual file but more crucially the .git/lfs/objects relevant file has become corrupt.

As a result, fetch or pull fail with the error from the original author.

The error I've traced back to `vendor/_nuts/github.com/tecnoweenie/go-contentaddressable/file.go` at the line `tempFIle,err := os.OpenFile(tempFilename, os.O_WRONLY|os.CREAET|os.O_EXCL, 0644)`

If I print err, the contents are (0xd64558, 0xc8200de600)
